capstone/src/test/java/com/google/sps/RecommenderAlgorithmTest.java

```diff
@@ -0,0 +1,422 @@
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.google.sps.service.DatabaseService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+ 
+@RunWith(JUnit4.class)
+public final class RecommenderAlgorithmTest {
+    
+    private final LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    private static final List<Tag> TEST_CATEGORY_TAGS = Tag.CATEGORY_TAGS;
+
+    private static User TEST_USER_SMITH;
+    private static User TEST_USER_JOHN;
+    private static User TEST_USER_BILL;
+    private static User TEST_USER_SAM;
+    private static User TEST_USER_ADAM;
+    private static Room TEST_ROOM_POETRY;
+    private static Room TEST_ROOM_RUN;
+    private static Room TEST_ROOM_COOK; 
+    private static Room TEST_ROOM_COOK_TWO;
+    private static Room TEST_ROOM_SCIENCE;
+
+    private static double epsilon = 0.00001;
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+        TEST_USER_SMITH = new User("1.0234", "Smith");
+        TEST_USER_JOHN = new User("2341.0", "John");
+        TEST_USER_BILL = new User("341.02", "Bill");
+        TEST_USER_SAM = new User("41.023", "Sam");
+        TEST_USER_ADAM = new User("521.13", "Adam");
+        DatabaseService.save(TEST_USER_SMITH.getUserEntity());
+        DatabaseService.save(TEST_USER_JOHN.getUserEntity());
+        DatabaseService.save(TEST_USER_BILL.getUserEntity());
+        DatabaseService.save(TEST_USER_SAM.getUserEntity());
+        DatabaseService.save(TEST_USER_ADAM.getUserEntity());
+ 
+        TEST_ROOM_POETRY = new Room(TEST_USER_SMITH, "How to Teach Poetry", "Learn how to teach Poetry");
+        TEST_ROOM_RUN = new Room(TEST_USER_JOHN, "Run Efficiently", "Improve your running experience");
+        TEST_ROOM_COOK = new Room(TEST_USER_BILL, "Iron Chef Master Course", "Cook Better");
+        TEST_ROOM_COOK_TWO = new Room(TEST_USER_SAM, "Iron Chef Master Course II", "Cook Better Part 2");
+        TEST_ROOM_SCIENCE = new Room(TEST_USER_ADAM, "Physics I", "Understand the world");
+        DatabaseService.save(TEST_ROOM_POETRY.getRoomEntity());
+        DatabaseService.save(TEST_ROOM_RUN.getRoomEntity());
+        DatabaseService.save(TEST_ROOM_COOK.getRoomEntity());
+        DatabaseService.save(TEST_ROOM_COOK_TWO.getRoomEntity());
+        DatabaseService.save(TEST_ROOM_SCIENCE.getRoomEntity());
+    }
+ 
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void calculateDistanceSameUserTest() { 
+        TEST_ROOM_RUN.addTag(Tag.FITNESS);
+        TEST_ROOM_RUN.addFollower(TEST_USER_SMITH);
```

Comment:
@707090 It was a really good call to update the cached interest vector when we add the follower to the room since here, all we need to call is addFollower() and check the cached interest vector for the tests.